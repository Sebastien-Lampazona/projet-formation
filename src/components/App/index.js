import { useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { useRoutes } from 'react-router-dom';
import * as sockets from 'src/features/sockets';
import routes from 'src/routes';

export default function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(sockets.init());
  }, []);

  const router = useRoutes(routes);
  return router;
}
