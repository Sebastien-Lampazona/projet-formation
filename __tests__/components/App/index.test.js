import { screen } from '@testing-library/react';
import App from 'src/components/App';
import { renderWithProviders } from '__tests__/utils';

describe('App Component', () => {
  it('affiche "BeerList" sur la page d\'accueil', () => {
    const { asFragment} = renderWithProviders(<App />);
    const linkElement = screen.getByText(/BeerList/i);

    expect(asFragment()).toMatchSnapshot();
    expect(linkElement).toBeInTheDocument();
  });
});
