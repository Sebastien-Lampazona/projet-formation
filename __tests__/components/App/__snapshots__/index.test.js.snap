// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component affiche "BeerList" sur la page d'accueil 1`] = `
<DocumentFragment>
  <main
    class="container"
  >
    <div
      class="beerlist-container"
    >
      <main>
        <h2>
          BeerList
        </h2>
        <h3>
          C'est de la super bi√®re
        </h3>
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-ixblex"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-ixblex"
          >
            <div>
              <div
                aria-busy="true"
                aria-live="polite"
                class="ant-spin ant-spin-spinning css-dev-only-do-not-override-ixblex"
              >
                <span
                  class="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                  <i
                    class="ant-spin-dot-item"
                  />
                </span>
              </div>
            </div>
            <div
              class="ant-spin-container ant-spin-blur"
            >
              <div
                class="ant-table ant-table-empty"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: fixed;"
                    >
                      <colgroup>
                        <col
                          style="width: 200px;"
                        />
                        <col />
                        <col />
                        <col />
                        <col
                          style="width: 100px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell ant-table-cell-ellipsis"
                            scope="col"
                            title="Nom de la bi√®re"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Nom de la bi√®re
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell ant-table-cell-ellipsis"
                            scope="col"
                            title="Tagline"
                          >
                            Tagline
                          </th>
                          <th
                            class="ant-table-cell ant-table-cell-ellipsis"
                            scope="col"
                            title="Description"
                          >
                            Description
                          </th>
                          <th
                            class="ant-table-cell ant-table-cell-ellipsis"
                            scope="col"
                            title="Tips"
                          >
                            Tips
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            D√©tail
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-placeholder"
                        >
                          <td
                            class="ant-table-cell"
                            colspan="5"
                          >
                            <div
                              class="css-dev-only-do-not-override-ixblex ant-empty ant-empty-normal"
                            >
                              <div
                                class="ant-empty-image"
                              >
                                <svg
                                  height="41"
                                  viewBox="0 0 64 41"
                                  width="64"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <g
                                    fill="none"
                                    fill-rule="evenodd"
                                    transform="translate(0 1)"
                                  >
                                    <ellipse
                                      cx="32"
                                      cy="33"
                                      fill="#f5f5f5"
                                      rx="32"
                                      ry="7"
                                    />
                                    <g
                                      fill-rule="nonzero"
                                      stroke="#d9d9d9"
                                    >
                                      <path
                                        d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                      />
                                      <path
                                        d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                        fill="#fafafa"
                                      />
                                    </g>
                                  </g>
                                </svg>
                              </div>
                              <div
                                class="ant-empty-description"
                              >
                                No data
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
              <ul
                class="ant-pagination ant-table-pagination ant-table-pagination-right css-dev-only-do-not-override-ixblex"
              >
                <li
                  class="ant-pagination-total-text"
                >
                  1-10 
                  <span
                    class="fetching-icon"
                  >
                    üç∫ 
                  </span>
                </li>
                <li
                  aria-disabled="true"
                  class="ant-pagination-prev ant-pagination-disabled"
                  title="Previous Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="left"
                      class="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                  tabindex="0"
                  title="1"
                >
                  <a
                    rel="nofollow"
                  >
                    1
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-2"
                  tabindex="0"
                  title="2"
                >
                  <a
                    rel="nofollow"
                  >
                    2
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-3"
                  tabindex="0"
                  title="3"
                >
                  <a
                    rel="nofollow"
                  >
                    3
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-4"
                  tabindex="0"
                  title="4"
                >
                  <a
                    rel="nofollow"
                  >
                    4
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
                  tabindex="0"
                  title="5"
                >
                  <a
                    rel="nofollow"
                  >
                    5
                  </a>
                </li>
                <li
                  class="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
                  tabindex="0"
                  title="Next 5 Pages"
                >
                  <a
                    class="ant-pagination-item-link"
                  >
                    <div
                      class="ant-pagination-item-container"
                    >
                      <span
                        aria-label="double-right"
                        class="anticon anticon-double-right ant-pagination-item-link-icon"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="double-right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                          />
                        </svg>
                      </span>
                      <span
                        class="ant-pagination-item-ellipsis"
                      >
                        ‚Ä¢‚Ä¢‚Ä¢
                      </span>
                    </div>
                  </a>
                </li>
                <li
                  class="ant-pagination-item ant-pagination-item-33"
                  tabindex="0"
                  title="33"
                >
                  <a
                    rel="nofollow"
                  >
                    33
                  </a>
                </li>
                <li
                  aria-disabled="false"
                  class="ant-pagination-next"
                  tabindex="0"
                  title="Next Page"
                >
                  <button
                    class="ant-pagination-item-link"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      aria-label="right"
                      class="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </li>
                <li
                  class="ant-pagination-options"
                >
                  <div
                    aria-label="Page Size"
                    class="ant-select ant-pagination-options-size-changer css-dev-only-do-not-override-ixblex ant-select-single ant-select-show-arrow"
                  >
                    <div
                      class="ant-select-selector"
                    >
                      <span
                        class="ant-select-selection-search"
                      >
                        <input
                          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-label="Page Size"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                      </span>
                      <span
                        class="ant-select-selection-item"
                        title="10 / page"
                      >
                        10 / page
                      </span>
                    </div>
                    <span
                      aria-hidden="true"
                      class="ant-select-arrow"
                      style="user-select: none;"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-select-suffix"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </main>
    </div>
  </main>
</DocumentFragment>
`;
